src/outputs/restart.cpp

```diff
@@ -148,330 +145,304 @@ void RestartOutput::WriteOutputFile(Mesh *pm, ParameterInput *pin, SimTime *tm)
+      hid_t vLocalSpace, vGlobalSpace;
+      H5S vLocalSpaceNew, vGlobalSpaceNew;
+      auto &mb = *(pm->block_list.front());
+      const hsize_t vlen = vwrite->GetDim(4);
+      local_count[4] = global_count[4] = vlen;
+      std::vector<Real> tmpData(varSize * vlen * num_blocks_local);
+
+      // create spaces if required
+      if (vlen == 1) {
+        vLocalSpace = local_DSpace;
+        vGlobalSpace = global_DSpace;
+      } else {
+        vLocalSpace = vLocalSpaceNew =
+            H5S::FromHIDCheck(H5Screate_simple(5, local_count, NULL));
+        vGlobalSpace = vGlobalSpaceNew =
+            H5S::FromHIDCheck(H5Screate_simple(5, global_count, NULL));
+      }
 
-    // load up data
-    hsize_t index = 0;
-    for (auto &pmb : pm->block_list) {
-      bool found = false;
-      auto ci = ContainerIterator<Real>(pmb->real_containers.Get(), m, true);
-      for (auto &v : ci.vars) {
-        // Note index 4 transposed to interior
-        if (vWriteName.compare(v->label()) == 0) {
-          auto v_h = v->data.GetHostMirrorAndCopy();
-          LOADVARIABLEONE(index, tmpData.data(), v_h, out_ib.s, out_ib.e, out_jb.s,
-                          out_jb.e, out_kb.s, out_kb.e, vlen);
-          found = true;
-          break;
+      // load up data
+      hsize_t index = 0;
+      for (auto &pmb : pm->block_list) {
+        bool found = false;
+        auto ci = ContainerIterator<Real>(pmb->real_containers.Get(), m, true);
+        for (auto &v : ci.vars) {
+          // Note index 4 transposed to interior
+          if (vWriteName.compare(v->label()) == 0) {
+            auto v_h = v->data.GetHostMirrorAndCopy();
+            LOADVARIABLEONE(index, tmpData.data(), v_h, out_ib.s, out_ib.e, out_jb.s,
+                            out_jb.e, out_kb.s, out_kb.e, vlen);
+            found = true;
+            break;
+          }
+        }
+        if (!found) {
+          std::stringstream msg;
+          msg << "### ERROR: Unable to find variable " << vWriteName << std::endl;
+          PARTHENON_FAIL(msg);
         }
       }
-      if (!found) {
-        std::stringstream msg;
-        msg << "### ERROR: Unable to find variable " << vWriteName << std::endl;
-        PARTHENON_FAIL(msg);
-      }
-    }
-    // write dataset to file
-    WRITEH5SLAB2(vWriteName.c_str(), tmpData.data(), file, local_start, local_count,
-                 vLocalSpace, vGlobalSpace, property_list);
-    if (vlen > 1) {
-      H5Sclose(vLocalSpace);
-      H5Sclose(vGlobalSpace);
+      // write dataset to file
+      WRITEH5SLAB2(vWriteName.c_str(), tmpData.data(), file, local_start, local_count,
+                   vLocalSpace, vGlobalSpace, property_list);
     }
   }
-  // close persistent data spaces
-  H5Sclose(local_DSpace);
-  H5Sclose(global_DSpace);
-
-#ifdef MPI_PARALLEL
-  /* release the file access template */
-  ierr = H5Pclose(acc_file);
-  ierr = MPI_Info_free(&FILE_INFO_TEMPLATE);
-#endif
-
-  H5Pclose(property_list);
-  H5Fclose(file);
```

Comment:
Are these now automatically closed due to the different scoping?

Reply:
That's right - all HDF5 handles are now closed automatically.

Reply:
There's a one-off type that frees the MPI_Info, too.